src/libvirtApi/storagePool.ts

```diff
@@ -72,77 +96,99 @@ export async function storagePoolGet({
+        store.dispatch(updateOrAddStoragePool(Object.assign({}, dumpxmlParams, props), !!updateOnly));
     } catch (ex) {
-        console.warn('GET_STORAGE_POOL action failed for path', objPath, ex.toString());
+        if (ex instanceof Error)
+            console.warn('GET_STORAGE_POOL action failed for path', objPath, ex.toString());
     }
 }
 
 export async function storagePoolGetAll({
     connectionName,
-}) {
+} : {
+    connectionName: ConnectionName,
+}): Promise<void[]> {
     try {
-        const [objPaths] = await call(connectionName, '/org/libvirt/QEMU', 'org.libvirt.Connect', 'ListStoragePools', [0],
-                                      { timeout, type: 'u' });
+        const [objPaths]: [string[]] = await call(
+            connectionName, '/org/libvirt/QEMU', 'org.libvirt.Connect', 'ListStoragePools', [0],
+            { timeout, type: 'u' });
         return await Promise.all(objPaths.map(async path => {
-            const [active] = await call(connectionName, path, 'org.freedesktop.DBus.Properties', 'Get',
-                                        ['org.libvirt.StoragePool', 'Active'], { timeout, type: 'ss' });
-            if (active.v)
+            const [active] = await call<[cockpit.Variant]>(connectionName, path, 'org.freedesktop.DBus.Properties', 'Get',
+                                                           ['org.libvirt.StoragePool', 'Active'], { timeout, type: 'ss' });
+            if (get_variant_boolean(active))
                 return storagePoolRefresh({ connectionName, objPath: path });
             else
                 return storagePoolGet({ connectionName, id: path });
         }));
     } catch (ex) {
-        console.warn('GET_ALL_STORAGE_POOLS action failed:', ex.toString());
+        if (ex instanceof Error)
+            console.warn('GET_ALL_STORAGE_POOLS action failed:', ex.toString());
         throw ex;
     }
 }
 
-export async function storagePoolGetCapabilities({ connectionName }) {
+export async function storagePoolGetCapabilities({ connectionName } : { connectionName: ConnectionName }): Promise<StoragePoolCapabilites> {
     // TODO: replace with D-Bus API once available https://issues.redhat.com/browse/RHEL-7045
-    const opts = { err: "message", environ: ['LC_ALL=C.UTF-8'] };
-    if (connectionName === 'system')
-        opts.superuser = 'try';
-
     try {
         const poolCapabilities = await cockpit.spawn(
             ["virsh", "-c", "qemu:///" + connectionName, "pool-capabilities"],
-            opts);
+            {
+                err: "message",
+                environ: ['LC_ALL=C.UTF-8'],
+                ...(connectionName === "system" ? { superuser: "try" } : { })
+            });
         return parsePoolCapabilities(poolCapabilities);
     } catch (ex) {
-        console.warn('virsh pool-capabilities failed:', ex.toString());
+        if (ex instanceof Error)
+            console.warn('virsh pool-capabilities failed:', ex.toString());
```

Comment:
Do we check if it's Error just for the `toString()` to work?

If so we could just do `ex?.toString()` instead

Reply:
I changed this all to `String(ex)`